ui/lib/CancelStream.tsx

```diff
@@ -1,7 +1,116 @@
-import * as React from "react";
+import React, { useState, useEffect } from "react";
+import { Stream } from "../../sdk/js/index";
+import { formatTime } from "./utils";
 
-const CancelStream = () => {
-  return <div />;
-};
+interface CancelStreamProps {
+  stream: Stream;
+  onComplete: (error: string | boolean) => void;
```

Comment:
Would this work?
```suggestion
  onComplete: (error: Error | null) => void;
```

Also. Can the component not display its own error?

Reply:
No we can't have it as null as it would make it impossible to check in the parent component (cf. closing the modal)

Reply:
Yes the component (update and cancel) can display their own error

Reply:
Edited so that the component displays its own error and we pass True to the parent component when stream is effectively being cancelled/updated

Reply:
why is the boolean needed? it's called `onComplete` so I'm assuming the context is that it completed (with or without an error)

Reply:
The boolean is needed to set a state in the parent component; so that we'll have something like displayModal={onComplete}

Reply:
oh ok I see what you meant

Reply:
updated